addons/test_mail/tests/test_mail_gateway.py

```diff
@@ -1728,6 +1729,32 @@ def test_message_hebrew_iso8859_8_i(self):
             content
         )
 
+    def test_message_windows_874(self):
+        # Email for Thai customers who use Microsoft email service.
+        # The charset is windows-874 which isn't natively supported by
+        # python, check that Odoo is still capable of decoding it.
+        # windows-874 is the Microsoft equivalent of cp874.
+        charset = 'windows-874'
+        subject = 'ขออนุญาตทบทวนเอกสารโครงการ'
+        encoded_subject = Header(subject, charset)
+        content = 'แมวสีดำเป็นสัตว์เลี้ยงที่น่ารักมากๆ แ=ละมีเสน่ห์ในตัวอย่างมาก ๆ ด้วย'
+        encoded_content = base64.b64encode(content.encode(charset)).decode()
```

Comment:
with some more time thinking about it I no longer think that it is a good idea to use `Header` and `content.encode(charset)`. We should provide with the actual bytes found in the customer's email (ask for their EML) and not engineer our way to reproduce such strings.

Here the problem is that we added that `windows-874` alias and we are using it to both encode + decode the text. We could replace `charset = 'windows-874'` by `charset = 'utf-8'` the test would green.

In the original test (the hebrew one) we shouldn't had use `encoded_content = base64.b64encode(content.encode(charset)).decode()` in the first place but instead provide with actual bytes extracted out of a real-world email. This `encoded_content` is testing when the encoded text is found in the body of the email / inside an attachment, I think we had no real world example at hand and decided to craft one ourself.

Reply:
So I suggest we put our hand on the EML file that doesn't work inside odoo (you can ask the help of the tech sm squad, they are used to this kind of situation) and base our test on that file.